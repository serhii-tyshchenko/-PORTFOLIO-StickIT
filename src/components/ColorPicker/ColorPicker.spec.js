/* eslint-disable react/jsx-props-no-spreading */
import renderer from 'react-test-renderer';

import { ColorPicker } from './index';

describe('ColorPicker', () => {
  it('renders correctly closed state', () => {
    const props = {
      onChange: jest.fn(),
      color: 'red',
      opened: false,
    };
    const component = renderer.create(<ColorPicker {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('renders correctly open state', () => {
    const props = {
      onChange: jest.fn(),
      color: 'red',
      opened: true,
    };
    const component = renderer.create(<ColorPicker {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
