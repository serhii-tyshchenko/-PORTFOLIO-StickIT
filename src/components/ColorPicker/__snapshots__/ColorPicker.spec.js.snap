// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders correctly closed state 1`] = `null`;

exports[`ColorPicker renders correctly open state 1`] = `
<div
  className="color-picker"
>
  <label
    className="color-picker__item"
    htmlFor="color-0"
    style={
      Object {
        "backgroundColor": "#FCCB00",
      }
    }
    title="#FCCB00"
  >
    <input
      checked={false}
      id="color-0"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#FCCB00"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-1"
    style={
      Object {
        "backgroundColor": "#EB9694",
      }
    }
    title="#EB9694"
  >
    <input
      checked={false}
      id="color-1"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#EB9694"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-2"
    style={
      Object {
        "backgroundColor": "#FAD0C3",
      }
    }
    title="#FAD0C3"
  >
    <input
      checked={false}
      id="color-2"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#FAD0C3"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-3"
    style={
      Object {
        "backgroundColor": "#FEF3BD",
      }
    }
    title="#FEF3BD"
  >
    <input
      checked={false}
      id="color-3"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#FEF3BD"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-4"
    style={
      Object {
        "backgroundColor": "#C1E1C5",
      }
    }
    title="#C1E1C5"
  >
    <input
      checked={false}
      id="color-4"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#C1E1C5"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-5"
    style={
      Object {
        "backgroundColor": "#BEDADC",
      }
    }
    title="#BEDADC"
  >
    <input
      checked={false}
      id="color-5"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#BEDADC"
    />
  </label>
  <label
    className="color-picker__item"
    htmlFor="color-6"
    style={
      Object {
        "backgroundColor": "#C4DEF6",
      }
    }
    title="#C4DEF6"
  >
    <input
      checked={false}
      id="color-6"
      name="colors"
      onChange={[Function]}
      type="radio"
      value="#C4DEF6"
    />
  </label>
</div>
`;
